### PR TITLE
Skip progress display when in non-terminal 

### PR DESCRIPTION
**Pull request by nicoddemus**
Similar to pytest-dev/pytest#1397

### CODE REVIEW

**Comment by nicoddemus** [comment]
Dropped pytest 2.4 and 2.5 from build Matrix and added pytest 2.9.

Question: do we really need to support such old pytest versions? I would think it reasonable to support only the most current and previous version, i. e. 2.8 and 2.9 currently.

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt any idea why the test suite is failing in pytest-2.9?

```
=================================== FAILURES ===================================
____________________________ test_remoteinitconfig _____________________________
testdir = <Testdir local('/tmp/pytest-of-travis/pytest-0/testdir/test_remoteinitconfig0')>
    def test_remoteinitconfig(testdir):
        from xdist.remote import remote_initconfig
        config1 = testdir.parseconfig()
        config2 = remote_initconfig(config1.option.__dict__, config1.args)
>       assert config2.option.__dict__ == config1.option.__dict__
E       assert {'assertmode'...': False, ...} == {'assertmode':...': False, ...}
E         Omitting 56 identical items, use -v to show
E         Differing items:
E         {'file_or_dir': ['/tmp/pytest-of-travis/pytest-0/testdir/test_remoteinitconfig0']} != {'file_or_dir': []}
E         Use -v to get the full diff
/home/travis/build/pytest-dev/pytest-xdist/testing/test_remote.py:69: AssertionError
```

File: xdist/dsession.py [hunk 1]
```diff
@@ -722,17 +722,20 @@ def __init__(self, config):
         self.tr = config.pluginmanager.getplugin("terminalreporter")
         self._status = {}
         self._lastlen = 0
+        self._isatty = self.tr.hasmarkup
```
Comment (RonnyPfannschmidt) [inline xdist/dsession.py]: on first gnalce i think that instead of hasattr, we should use `getattr(self.tr, 'isatty', self.tr.hasmarkup)`

**Comment by RonnyPfannschmidt** [comment]
off hand no idea

**Comment by nicoddemus** [comment]
OK thanks, I will investigate later, although it seems this is broken in `master` already because it seems unrelated to my change.

**Comment by RonnyPfannschmidt** [comment]
i would guess its related to the change @bukzor (?) did to fix wrong behavior wrt xdist context initialization

**Comment by nicoddemus** [comment]
It's my hunch as well

**Comment by RonnyPfannschmidt** [comment]
please xfail it and report an issue where we can solve it

**Comment by nicoddemus** [comment]
Will do :grin:

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt created #59 as we agreed.

Let me know your thoughts on this patch when you can. :grin:

**Comment by RonnyPfannschmidt** [comment]
thanks :+1:

**Comment by nicoddemus** [comment]
Thank you. I will take a look at #59 later.

**Pull request merged**